Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+    connected: str
+    hostname: str
+    last_commit_all_state_sp: str
+    name: str = ""
+
+    _output_prefix = OUTPUT_PREFIX + "DeviceGroupOp"
+    _title = "PAN-OS Operational Device Group Status"
+    _outputs_key_field = "name"
+
+
+@dataclass
+class TemplateStackInformation(ResultData):
+    """
+    :param serial: Serial number of firewall
+    :param connected: Whether the firewall is currently connected
+    :param hostname: Firewall hostname
+    :param last_commit_all_state_tpl: Text state of last commit
+    :param name: Template Stack Name
+    """
+    serial: str
+    connected: str
+    hostname: str
+    last_commit_all_state_tpl: str
+    name: str = ""
+
+    _output_prefix = OUTPUT_PREFIX + "TemplateStackOp"
+    _title = "PAN-OS Operational Template Stack status"
+    _outputs_key_field = "name"
+
+
+@dataclass
+class PanosObjectReference(ResultData):
+    """
+    :param container_name: What parent container (DG, Template, VSYS) this object belongs to.
+    :param name: The PAN-OS object name
+    :param object_type: The PAN-OS-Python object type
+    """
+    container_name: str
+    name: str
+    object_type: str
+
+    _output_prefix = OUTPUT_PREFIX + "PanosObject"
+    _title = "PAN-OS Objects"
+
+
+def dataclass_from_dict(device: Union[Panorama, Firewall], d: dict, class_type: Callable):
+    """
+    Given a dictionary and a datacalass, converts the dictionary into the dataclass type.
+    """
+    if device.hostname:
+        d["hostid"] = device.hostname
+    if device.serial:
+        d["hostid"] = device.serial
+
+    r = {}
+    for k, v in d.items():
+        d_key = k.replace("-", "_")
+        dataclass_field = next((x for x in fields(class_type) if x.name == d_key), None)
+        if dataclass_field:
+            r[d_key] = v
+
+    return class_type(**r)
+
+
+def flatten_xml_to_dict(element: Element, d: dict, class_type: Callable):
+    """Given an XML element, a dictionary, and a class, flattens the XML into the class."""
+    for child_element in element:
+        tag = child_element.tag
+        # Replace hyphens in tags with underscores to match python attributes
+        tag = tag.replace("-", "_")
+        dataclass_field = next((x for x in fields(class_type) if x.name == tag), None)
+        if dataclass_field:
+            d[tag] = child_element.text
+
+        if len(child_element) > 0:
+            d = {**d, **flatten_xml_to_dict(child_element, d, class_type)}
+
+    return d
+
+
+def dataclass_from_element(
+        device: Union[Panorama, Firewall],
+        class_type: Callable,
+        element: Optional[Element],
+):
+    """
+    Turns an XML `Element` Object into an instance of the provided dataclass. Dataclass paramaters must match
+    child XML tags exactly.
+    """
+    d = {}
+    if element is None:
+        return
+
+    if device.hostname:
+        d["hostid"] = device.hostname
+    if device.serial:
+        d["hostid"] = device.serial
+
+    child_element: Element
```

Comment:
i might be missing something, but i don't see it being used anywhere

Reply:
true - removed